clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -237,6 +245,72 @@ class BuiltinNameEmitter {
   // same entry (<I1, I2, I3>).
   MapVector<BuiltinIndexListTy *, BuiltinTableEntries> SignatureListMap;
 };
+
+/// Generate a header file representing the builtins as understood by clang.
+/// This allow the generation of a library interface.
+class JSONBuiltinInterfaceEmitter : public BuiltinNameEmitter {
+public:
+  struct TypeDesc {
+    TypeDesc(const Record *T);
+    // Return a string representing the "base" type (ignores pointers).
+    std::string GetBaseTypeAsStr() const;
+
+    const Record *Rec;
```

Comment:
Since you decompose basically the whole record, why do you need it here as well?

Reply:
That was for debug, removed